apps/marketplace/pages/BuyerSpecialistFlowPage.js

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { actions } from 'react-redux-form'
+import { Redirect } from 'react-router-dom'
+import formProps from 'shared/form/formPropsSelector'
+import { ErrorBoxComponent } from 'shared/form/ErrorBox'
+import ProgressFlow from 'marketplace/components/ProgressFlow/ProgressFlow'
+import BuyerSpecialistStages from 'marketplace/components/BuyerSpecialist/BuyerSpecialistStages'
+import { rootPath } from 'marketplace/routes'
+import { loadPublicBrief, saveBrief } from 'marketplace/actions/briefActions'
+import { setErrorMessage } from 'marketplace/actions/appActions'
+import LoadingIndicatorFullPage from 'shared/LoadingIndicatorFullPage/LoadingIndicatorFullPage'
+import { BuyerSpecialistFormReducer } from 'marketplace/reducers'
+
+const model = 'BuyerSpecialistForm'
+
+export class BuyerSpecialistFlowPage extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      loading: false,
+      flowIsDone: false
+    }
+
+    this.saveBrief = this.saveBrief.bind(this)
+    this.handleStageMount = this.handleStageMount.bind(this)
+  }
+
+  componentDidMount() {
+    if (this.props.match.params.briefId) {
+      this.getBriefData()
+    }
+  }
+
+  getBriefData() {
+    this.setState({
+      loading: true
+    })
+    this.props.loadInitialData(this.props.match.params.briefId).then(response => {
+      // only accept data defined in the form reducer
+      const data = { ...BuyerSpecialistFormReducer }
+      if (response.data.brief) {
+        Object.keys(response.data.brief).map(property => {
+          if (Object.keys(BuyerSpecialistFormReducer).includes(property)) {
+            data[property] = response.data.brief[property]
+          }
+          return true
+        })
+
+        if (response.data.brief.status && response.data.brief.status !== 'draft') {
+          this.props.setError('You cannot edit this opportunity as you have already published it.')
```

Comment:
Would this message show if the opportunity was deleted / withdrawn?

Reply:
it does show, I don't think it is a problem though because it is mostly accurate.